src/xtgeo/grid3d/grid.py

```diff
@@ -132,6 +183,71 @@ def grid_from_roxar(
 IJKRange = Tuple[int, int, int, int, int, int]
 
 
+def allow_deprecated_init(func):
+    # This decorator is here to maintain backwards compatibility in the construction
+    # of RegularSurface and should be deleted once the deprecation period has expired,
+    # the construction will then follow the new pattern.
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        # Checking if we are doing an initialization
+        # from file and raise a deprecation warning if
+        # we are.
+        if "gfile" in kwargs or (
+            len(args) >= 1 and isinstance(args[0], (str, Path, _XTGeoFile))
+        ):
+            warnings.warn(
+                "Initializing directly from file name is deprecated and will be "
+                "removed in xtgeo version 4.0. Use: "
+                "mygrid = xtgeo.grid_from_file('some_name.roff') instead",
+                DeprecationWarning,
+            )
+
+            def constructor(**kwargs):
+                func(self, **kwargs)
+                return self
+
+            _handle_import(constructor, *args, **kwargs)
+            return None
+
+        # Check if we are doing default value init
+        if len(args) == 0 and len(kwargs) == 0:
+            warnings.warn(
+                "Initializing default box grid is deprecated and will be "
+                "removed in xtgeo version 4.0. Use: "
+                "mygrid = xtgeo.create_box_grid() or, alternatively,"
+                "directly from file with mygrid = xtgeo.grid_from_file().",
+                DeprecationWarning,
+            )
+            kwargs = _grid_etc1.create_box(dimension=(4, 3, 5))
+            return func(self, **kwargs)
+        return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+def allow_xtgformat1_init(func):
+    # This decorator is here to maintain backwards compatibility with
+    # initializing grid with xtgformat=1 which some internal functionality
+    # still does. This is only supported internally.
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+        # Checking if we are doing an initialization
+        # from file and raise a deprecation warning if
+        # we are.
```

Comment:
This is just a copy paste thingy right? We're not going to raise deprecation warnings on this one? Do we know how long this decorator is ment to stay?

Reply:
Indeed, removed. We want to remove xtgformat=1 anyways, so I guess it will stay only as long as we keep producing xtgformat=1 grids, which is not exposed to the user. We can likely remove this rather quickly.